### PR TITLE
[SEDONA-364] Add RS_MinConvexHull

### PR DESCRIPTION
**Pull request by iGN5117**
## Did you read the Contributor Guide?

- Yes, I have read [Contributor Rules](https://sedona.apache.org/latest-snapshot/community/rule/) and [Contributor Development Guide](https://sedona.apache.org/latest-snapshot/community/develop/)

## Is this PR related to a JIRA ticket?

- Yes, the URL of the associated JIRA ticket is https://issues.apache.org/jira/browse/SEDONA-364. The PR name follows the format `[SEDONA-XXX] my subject`.

## What changes were proposed in this PR?
- Add RS_MinConvexHull

## How was this patch tested?
- Added new tests

## Did this PR include necessary documentation updates?

- Yes, I am adding a new API. I am using the [current SNAPSHOT version number](https://github.com/apache/sedona/blob/master/pom.xml#L29) in since `vX.Y.Z` format.

### CODE REVIEW

File: common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java [hunk 1]
```diff
@@ -61,6 +63,70 @@ public static Geometry convexHull(GridCoverage2D raster) throws FactoryException
...
+        if (!allBands) ensureSafeBand(raster, band);
+        int width = RasterAccessors.getWidth(raster), height = RasterAccessors.getHeight(raster), srid = RasterAccessors.srid(raster);
+        int minX = Integer.MAX_VALUE, maxX = Integer.MIN_VALUE, minY = Integer.MAX_VALUE, maxY = Integer.MIN_VALUE;
+        for (int j = 0; j < height; j++) {
+            for (int i = 0; i < width; i++) {
+                // get the value of the raster at the coordinates i, j
+                //if value is not no data, update variables to track minX, maxX, minY, maxY
+                GridCoordinates2D currGridCoordinate = new GridCoordinates2D(i, j);
+                double[] bandPixelValues = raster.evaluate(currGridCoordinate, (double[]) null);
+                if (allBands) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java:77]: The duplicate code here can be merged. You can use the same for loop for both condition by manually controlling the start and end of this loop.

File: common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java [hunk 2]
```diff
@@ -76,6 +77,49 @@ public void testConvexHullFromRasterFile() throws FactoryException, TransformExc
         assertEquals(expectedCoordOne.getY(), coordinates[4].getY(), 0.5d);
     }
 
+    @Test
+    public void testMinConvexHull() throws FactoryException, TransformException {
```
Comment (jiayuasu) [inline common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java:81]: Do these tests yield the same results as PostGIS?

File: docs/api/sql/Raster-operators.md [hunk 3]
```diff
@@ -62,14 +62,70 @@ Since: `1.5.0`
...
 ```
 
 Output:
 ```
 POLYGON ((156 -132, 181 -107, 211 -7, 186 -32, 156 -132))
 ```
 
+### RS_MinConvexHull
+
+Introduction: Returns the min convex hull geometry of the raster **excluding** the NoDataBandValue band pixels, in the given band.
```
Comment (jiayuasu) [inline docs/api/sql/Raster-operators.md:75]: Please also add that the produced geometry uses the world coordinate in the CRS specified by the raster.

File: common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java [hunk 4]
```diff
@@ -61,6 +63,70 @@ public static Geometry convexHull(GridCoverage2D raster) throws FactoryException
...
+                    for (int currBand = 1; currBand <= RasterAccessors.numBands(raster); currBand++) {
+                        double currBandValue = bandPixelValues[currBand - 1];
+                        Double bandNoDataValue = RasterBandAccessors.getBandNoDataValue(raster, currBand);
+                        if ((bandNoDataValue == null) || currBandValue != bandNoDataValue) {
```
Comment (jiayuasu) [inline common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java:81]: When there are multiple bands, is the overall minConvexHull the hull that has the minX, minY, maxX, maxY among all bands?

**Review by jiayuasu — CHANGES_REQUESTED** [review]

File: common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java (hunk 2 above)
Comment (iGN5117) [reply common/src/test/java/org/apache/sedona/common/raster/GeometryFunctionsTest.java:81]: Yes, all tests are done both on sedona and PostGIS

File: docs/api/sql/Raster-operators.md (hunk 3 above)
Comment (iGN5117) [reply docs/api/sql/Raster-operators.md:75]: Good point, I added this in the docs

File: common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java (hunk 1 above)
Comment (iGN5117) [reply common/src/main/java/org/apache/sedona/common/raster/GeometryFunctions.java]: Great suggestion! I've refactored the code and pushed

**Review by jiayuasu — APPROVED** [review]

**Pull request merged**